### PR TITLE
refactor(admin): switch from webpack to vite

### PR DESCRIPTION
**Pull request by mkue**
@renestalder, FYI: I removed the `@storybook/builder-webpack5` dependencies again from `ui/package.json`.

### CODE REVIEW

File: ui/README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ $ npm i @socialincome/ui
 Use the components:
 
 ```tsx
-import { SoButton } from '@socialincome/ui';
+
```
Comment (mkue) [inline ui/README.md:25]:
```suggestion
import { SoButton } from '@socialincome/ui';
```

**Comment by mkue** [comment]
> Visit the preview URL for this PR (updated for commit [53ccd73](https://github.com/socialincome-san/public/commit/53ccd737e35e95a969dc511c372d2a46392fdcba)):
> 
> https://admin-social-income-prod--pr214-mkue-admin-switch-to-ry1s1wbt.web.app
> 
> (expires Fri, 23 Dec 2022 22:09:37 GMT)
> 
> 🔥 via [Firebase Hosting GitHub Action](https://github.com/marketplace/actions/deploy-to-firebase-hosting) 🌎
> 
> Sign: f24fba79623410c6544bf7eccf94b222a68fadf8

@ssandino, this also fixes the refreshing issue we had. Unfortunately only in production yet. For the local setup I have to figure out still...

**Review by ssandino — APPROVED** [review]

File: .github/workflows/admin.yml [hunk 2]
```diff
@@ -43,15 +43,15 @@ jobs:
...
-          echo REACT_APP_ADMIN_FB_API_KEY=${{ secrets.REACT_APP_ADMIN_FB_API_KEY }} > .env
-          echo REACT_APP_ADMIN_FB_AUTH_DOMAIN=${{ secrets.REACT_APP_ADMIN_FB_AUTH_DOMAIN }} >> .env
-          echo REACT_APP_ADMIN_FB_DATABASE_URL=${{ secrets.REACT_APP_ADMIN_FB_DATABASE_URL }} >> .env
-          echo REACT_APP_ADMIN_FB_MEASUREMENT_ID=${{ secrets.REACT_APP_ADMIN_FB_MEASUREMENT_ID }} >> .env
-          echo REACT_APP_ADMIN_FB_MESSAGING_SENDER_ID=${{ secrets.REACT_APP_ADMIN_FB_MESSAGING_SENDER_ID }} >> .env
-          echo REACT_APP_ADMIN_FB_PROJECT_ID=${{ secrets.REACT_APP_ADMIN_FB_PROJECT_ID }} >> .env
-          echo REACT_APP_ADMIN_FB_STORAGE_BUCKET=${{ secrets.REACT_APP_ADMIN_FB_STORAGE_BUCKET }} >> .env
-          echo REACT_APP_ADMIN_ALGOLIA_APPLICATION_ID=${{ secrets.REACT_APP_ADMIN_ALGOLIA_APPLICATION_ID }} >> .env
-          echo REACT_APP_ADMIN_ALGOLIA_SEARCH_KEY=${{ secrets.REACT_APP_ADMIN_ALGOLIA_SEARCH_KEY }} >> .env
+          echo VITE_ADMIN_FB_API_KEY=${{ secrets.REACT_APP_ADMIN_FB_API_KEY }} > .env
```
Comment (andrashee) [inline .github/workflows/admin.yml:46]: Are the GitHub secrets now a bit inconsistent with the local / prod envs? Not a tragedy, though.

File: admin/src/collections/Users.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ export const usersCollection = buildCollection<User>({
 	}),
 	additionalColumns: [FirstNameCol, LastNameCol, GenderCol, PhoneCol, CountryCol, CityCol, ReferralCol],
 	subcollections: [contributionsCollection, donationCertificateCollection],
-	extraActions: CreateDonationCertificatesAction,
+	Actions: CreateDonationCertificatesAction,
```
Comment (andrashee) [inline admin/src/collections/Users.ts:92]: what's this change about?

File: admin/src/env.d.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+interface ImportMeta {
+	readonly env: {
+		VITE_ADMIN_FB_AUTH_EMULATOR_URL: string;
```
Comment (andrashee) [inline admin/src/env.d.ts:3]: I like it 👍. 
Wondering if the config.tsx still adds some additional value or is just one more indirection?

**Review by andrashee — APPROVED** [review]
Looks very nice. Thanks for going through this @mkue.

Haven't run it by myself, but judging from the console output of the Github builds, this change is huge:

Before:
```
WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
[34](https://github.com/socialincome-san/public/actions/runs/3719455403/jobs/6308324150#step:7:35)
Entrypoints:
[35](https://github.com/socialincome-san/public/actions/runs/3719455403/jobs/6308324150#step:7:36)
  main (39.3 MiB)
[36](https://github.com/socialincome-san/public/actions/runs/3719455403/jobs/6308324150#step:7:37)
      app.js
```

now:
```
vite v4.0.1 building for production...
[13](https://github.com/socialincome-san/public/actions/runs/3719518854/jobs/6308426507#step:7:14)
transforming...
[14](https://github.com/socialincome-san/public/actions/runs/3719518854/jobs/6308426507#step:7:15)
✓ 14037 modules transformed.
[15](https://github.com/socialincome-san/public/actions/runs/3719518854/jobs/6308426507#step:7:16)
rendering chunks...
[16](https://github.com/socialincome-san/public/actions/runs/3719518854/jobs/6308426507#step:7:17)
computing gzip size...
[17](https://github.com/socialincome-san/public/actions/runs/3719518854/jobs/6308426507#step:7:18)
dist/index.html                     0.74 kB
[18](https://github.com/socialincome-san/public/actions/runs/3719518854/jobs/6308426507#step:7:19)
dist/assets/index-1765c5a7.css     30.23 kB │ gzip:     5.46 kB
[19](https://github.com/socialincome-san/public/actions/runs/3719518854/jobs/6308426507#step:7:20)
dist/assets/index-5177e84e.js   7,474.56 kB │ gzip: 1,712.49 kB
[20](https://github.com/socialincome-san/public/actions/runs/3719518854/jobs/6308426507#step:7:21)
```

🥳

File: .github/workflows/admin.yml (hunk 2 above)
Comment (mkue) [reply .github/workflows/admin.yml:46]: Yeah. It's a little mess now, but still completely transparent and totally fine for me...

File: admin/src/collections/Users.ts (hunk 3 above)
Comment (mkue) [reply admin/src/collections/Users.ts:92]: Sandino wished to upgrade FireCMS, which did in this PR as well. The parameter name just changed, so I updated it.

File: admin/src/env.d.ts (hunk 4 above)
Comment (mkue) [reply admin/src/env.d.ts:3]: Very true. It's not really needed anymore. I'll clean it up.

**Pull request merged**